components/scale/dateTime.js

```diff
@@ -60,6 +60,35 @@
             return scale;
         };
 
+        scale.nice = function() {
+            adaptedScale.nice();
+            var domain = adaptedScale.domain();
+            var domainLower = discontinuities().clampUp(domain[0]);
+            var domainUpper = discontinuities().clampDown(domain[1]);
+            adaptedScale.domain([domainLower, domainUpper]);
+            return scale;
+        };
+
+        scale.ticks = function() {
+            var ticks = adaptedScale.ticks();
+            var clampedTicks = ticks.map(function(tick, index) {
+                if (index < ticks.length - 1) {
+                    return discontinuities().clampUp(tick);
+                } else {
+                    var clampedTick = discontinuities().clampUp(tick);
+                    return clampedTick < scale.domain()[1] ?
+                        clampedTick : discontinuities().clampDown(tick);
+                }
+            });
+            var uniqueTicks = clampedTicks.reduce(function(arr, tick) {
```

Comment:
Actually, doesn't work so well in this context, would require a map( d.getTime() ) before using d3.set, and an inverse mapping afterwards.